numpy/core/arrayprint.py

```diff
@@ -1220,15 +1220,20 @@ def format_float_positional(x, precision=None, unique=True,
 
 
 class IntegerFormat:
-    def __init__(self, data):
+    def __init__(self, data, sign=False):
+        if isinstance(sign, bool):
```

Comment:
Why do we accept a `bool` here? Because `sign` goes to both `IntegerFormat` and `FloatingFormat`, the accepted values need to be the same. They are exactly just those objects in the set `{None, '+', ' ', '-'}`.

Reply:
Ah, I see you got it from `FloatingFormat`, which accepts it due to backwards compatibility concerns. I don't think that will be an issue for `IntegerFormat`.

Reply:
Got rid of the `bool` check.